prediction_market_agent/agents/microchain_agent/app.py

```diff
@@ -57,7 +60,9 @@
 ALLOW_STOP = False
 
 
+@observe()
 def run_agent(agent: Agent, iterations: int, model: SupportedModel) -> None:
+    langfuse_context.update_current_trace(tags=[GENERAL_AGENT_TAG, STREAMLIT_TAG])
```

Comment:
Does it make more sense to run this once at the start of the app, e.g. just after `ENABLE_LANGFUSE` is defined, you could do
```
if ENABLE_LANGFUSE:
    langfuse_context.update_current_trace(tags=[GENERAL_AGENT_TAG, STREAMLIT_TAG])
```

Reply:
After the app is started, there is no observation yet. The first `@observe()` is on this `run_agent`.

However, it's not awesome that every click on "Run Agent" is a separate track in langfuse, but I at least added a session id, so they are grouped.